crates/ruff_python_parser/src/lexer.rs

```diff
@@ -1318,25 +1321,74 @@ impl<'source> Lexer<'source> {
     fn token_start(&self) -> TextSize {
         self.token_range().start()
     }
+
+    pub(crate) fn take_value(&mut self) -> TokenValue {
+        std::mem::replace(&mut self.value, TokenValue::None)
```

Comment:
Nit: You can use `std::mem::take` if you implement `Default` for `TokenValue`
```suggestion
        std::mem::take(&mut self.value)
```